### ISSUE TITLE
[BUG] Automated PDF imports are not working

### ISSUE THREAD

**Issue by DeltaTangoLima** [paperless-ngx/paperless-ngx#870]
### Description

I commented on #858, but the OP there was reporting a missing dependency in an unrelated docker image, so apologies if this is considered a dupe of that issue.

I'm having the same issue as #858 (can't tag/import PDFs), but mine's been since the latest update to the [official](https://github.com/paperless-ngx/paperless-ngx/pkgs/container/paperless-ngx) docker image.

Symptoms for me were two-fold:

1. I had one particular PDF that just wouldn't let me save changes to the tag (ie: remove the inbox tag). After a lot of repeated attempts, it eventually worked.
2. I've had 4 PDFs that simply won't complete the automated consumption process - a few of them via automated email processing and another from the consume directory that my scanner writes to.

I've made no other changes to what was a working system, except for updating to the latest docker image. Logs show that paperless-ngx seems to consume the file just fine, and even says it's tagging it, but it never appears in the paperless-ngx UI, and the file that my scanner wrote out stays in the consume directory.

Relevant log entries (sensitive tag data redacted) - I see the same pattern for files I've scanned to my consume directory and files that were processed via email, so unlikely to be a permissions issue on my host system.

```
[2022-05-03 04:30:05,374] [INFO] [paperless.management.consumer] Adding /usr/src/paperless/src/../consume/img-220503142929.pdf to the task queue.
[2022-05-03 04:30:07,303] [INFO] [paperless.consumer] Consuming img-220503142929.pdf
[2022-05-03 04:30:07,334] [DEBUG] [paperless.consumer] Detected mime type: application/pdf
[2022-05-03 04:30:07,350] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser
[2022-05-03 04:30:07,372] [DEBUG] [paperless.consumer] Parsing img-220503142929.pdf...
[2022-05-03 04:30:09,343] [DEBUG] [paperless.parsing.tesseract] Extracted text from PDF file /usr/src/paperless/src/../consume/img-220503142929.pdf
[2022-05-03 04:30:10,207] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': '/usr/src/paperless/src/../consume/img-220503142929.pdf', 'output_file': '/tmp/paperless/paperless-7ah5ge8t/archive.pdf', 'use_threads': True, 'jobs': 2, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': '/tmp/paperless/paperless-7ah5ge8t/sidecar.txt'}
[2022-05-03 04:33:49,129] [DEBUG] [paperless.parsing.tesseract] Using text from sidecar file
[2022-05-03 04:33:49,143] [DEBUG] [paperless.consumer] Generating thumbnail for img-220503142929.pdf...
[2022-05-03 04:33:49,190] [DEBUG] [paperless.parsing] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient /tmp/paperless/paperless-7ah5ge8t/archive.pdf[0] /tmp/paperless/paperless-7ah5ge8t/convert.png
[2022-05-03 04:34:10,522] [DEBUG] [paperless.parsing.tesseract] Execute: optipng -silent -o5 /tmp/paperless/paperless-7ah5ge8t/convert.png -out /tmp/paperless/paperless-7ah5ge8t/thumb_optipng.png
[2022-05-03 04:36:22,272] [DEBUG] [paperless.consumer] Saving record to database
[2022-05-03 04:36:22,495] [INFO] [paperless.handlers] Assigning correspondent [REDACTED CORRESPONDENT] to 2022-04-21 img-220503142929
[2022-05-03 04:36:22,567] [INFO] [paperless.handlers] Tagging "2022-04-21 [REDACTED CORRESPONDENT] img-220503142929" with "[REDACTED DOCUMENT TAG]"
```

Only other thing I can find that may be a clue is log entries in the database container that *may* correspond (timestamps don't line up) with each attempted write/tag of a file:

```
LOG:  unexpected EOF on client connection with an open transaction
```

Could there have been a breaking change resulting in issues writing to the database under specific conditions?

### Steps to reproduce

See above - problem is a result of automated consumption of PDF files

### Webserver logs

```bash
[2022-05-03 04:30:05,374] [INFO] [paperless.management.consumer] Adding /usr/src/paperless/src/../consume/img-220503142929.pdf to the task queue.
[2022-05-03 04:30:07,303] [INFO] [paperless.consumer] Consuming img-220503142929.pdf
[2022-05-03 04:30:07,334] [DEBUG] [paperless.consumer] Detected mime type: application/pdf
[2022-05-03 04:30:07,350] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser
[2022-05-03 04:30:07,372] [DEBUG] [paperless.consumer] Parsing img-220503142929.pdf...
[2022-05-03 04:30:09,343] [DEBUG] [paperless.parsing.tesseract] Extracted text from PDF file /usr/src/paperless/src/../consume/img-220503142929.pdf
[2022-05-03 04:30:10,207] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': '/usr/src/paperless/src/../consume/img-220503142929.pdf', 'output_file': '/tmp/paperless/paperless-7ah5ge8t/archive.pdf', 'use_threads': True, 'jobs': 2, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': '/tmp/paperless/paperless-7ah5ge8t/sidecar.txt'}
[2022-05-03 04:33:49,129] [DEBUG] [paperless.parsing.tesseract] Using text from sidecar file
[2022-05-03 04:33:49,143] [DEBUG] [paperless.consumer] Generating thumbnail for img-220503142929.pdf...
[2022-05-03 04:33:49,190] [DEBUG] [paperless.parsing] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient /tmp/paperless/paperless-7ah5ge8t/archive.pdf[0] /tmp/paperless/paperless-7ah5ge8t/convert.png
[2022-05-03 04:34:10,522] [DEBUG] [paperless.parsing.tesseract] Execute: optipng -silent -o5 /tmp/paperless/paperless-7ah5ge8t/convert.png -out /tmp/paperless/paperless-7ah5ge8t/thumb_optipng.png
[2022-05-03 04:36:22,272] [DEBUG] [paperless.consumer] Saving record to database
[2022-05-03 04:36:22,495] [INFO] [paperless.handlers] Assigning correspondent [REDACTED CORRESPONDENT] to 2022-04-21 img-220503142929
[2022-05-03 04:36:22,567] [INFO] [paperless.handlers] Tagging "2022-04-21 [REDACTED CORRESPONDENT] img-220503142929" with "[REDACTED DOCUMENT TAG]"
```

### Paperless-ngx version

1.7.0

### Host OS

Debian 10.9

### Installation method

Docker

### Browser

Firefox (unrelated)

**Comment by stumpylog** [comment]
The database log error seems related to me.  I'm assuming this is using a Postgres container for the DB?

**Comment by DeltaTangoLima** [comment]
> I'm assuming this is using a Postgres container for the DB?

Ah, yes - sorry.  I'm using what was the suggested docker-compose.yml from the paperless-ng project, so using ```postgres:13``` image (not :latest) and ```redis:6.0``` (again, not :latest).

**Comment by stumpylog** [comment]
You mentioned the document in the consume folder remains in place, is there a document copied into the media folder?  

ie, did the consumption otherwise happen, but the document just isn't in the db?

**Comment by DeltaTangoLima** [comment]
Good question - I just checked, and the most recent file in ```media/documents/originals``` is the last one that was successfully consumed, meaning it had the following log entry that the ones I mentioned above didn't:

```
[2022-04-30 23:55:59,319] [INFO] [paperless.consumer] Document 2022-04-29 [REDACTED CORRESPONDENT] 17746888 consumption finished
```

**Comment by stumpylog** [comment]
Very odd indeed.  And this is only happening with a subset of documents?  Which you can reproduce the issue with?

At best guess, something is crashing after the DB save line, which rolls back the transaction.  Except everything there should be caught and loudly reported.

**Comment by DeltaTangoLima** [comment]
Ah, sorry, no - not a subset of files. Just the 4 latest files since updating to the new docker image. Up until then, I'd succesfully ingested 1440 files into my setup. I just tried with a new file I had to scan yesterday, and had the same outcome.

I'll try the :dev image and, if that still doesn't work, I'll go back to the last stable release before current, and see how that goes.

**Comment by DeltaTangoLima** [comment]
*bump* any suggestions? There's definitely an issue writing to the database - I can't modify any tags or other metadata on any existing document already in the database.

**Comment by stumpylog** [comment]
I've thrown about 200 documents at a test instance with a postgress databse and haven't been able to reproduce this yet.  You're using a compose file [such as this one](https://github.com/paperless-ngx/paperless-ngx/blob/main/docker/compose/docker-compose.postgres.yml)?

Our database interface remained at the same version from 1.6.0 -> 1.7.0.  Nothing else but some migrations I can see.  So far, I'm at a loss.

**Comment by DeltaTangoLima** [comment]
My compose isn't a lot different - definitely the same images, my volume arrangement is tailored to my host, setup and I've put my environment into the compose itself (I use Portainer, so easier to manage that way).

I might move the db volume out of the way, create a fresh one and restore from one of my overnight backups. Maybe some perms got borked somehow...?

¯\\_(ツ)\_/¯

**Comment by stumpylog** [comment]
I might even consider exporting and recreate the database entirely. It's a bit of a nuclear option, but at least the ocr and other heavy processing won't need to run again

**Comment by DeltaTangoLima** [comment]
OK. I ```pg_dump```'d the database to a plaintext SQL file, stopped the broker & webserver containers, went into the db container and dropped/created an empty paperless db, then re-imported the SQL.

All my existing data is there, new docs still don't make it into the database.  Same log entries as before.

I'm at a loss on what to do next...

**Comment by DeltaTangoLima** [comment]
OK, weird. I think I got to the bottom of it, but not sure how. I realised I hadn't explicitly defined my UID and GID mappings in my Portainer compose. I've never had to before, and the volume I mount for my paperless-ngs media is on one of my NAS shares.

Anyway, just to completely eliminate the possibility, I mapped UID and GID 1000 in my compose, and it started working. This has me stumped - no users, groups or perms have changed at all.

Anyway, chalking it up to a layer 8 issue of some sort.

¯\\_(ツ)\_/¯
